forte/data/data_store.py

```diff
@@ -474,3 +476,79 @@ def prev_entry(self, tid: int) -> List:
             The previous entry of the same type as the ``tid`` entry.
         """
         raise NotImplementedError
+
+    def parse_onto_file(self):
+        r"""If user provides a customized ontology json file, we parse
+        this file and set self._type_attributes accordingly. It not only
+        includes the classes in the provided ontology file, but also import
+        classes from parent entry files.
+        """
+        if self.onto_file_path is None:
+            return
+
+        ontoGen = OntologyCodeGenerator()
+        merged_scheme, merged_dict = [], []
+        ontoGen.parse_ontology_spec(
+            self.onto_file_path, merged_scheme, merged_dict
+        )
+
+        self._type_attributes = {}
+
+        for onto in merged_scheme:
+            entry_name = onto["entry_name"]
+            attr_dic = {}
+            idx = 4
+            if "attributes" in onto:
+                for attr_dict in onto["attributes"]:
+                    name = attr_dict["name"]
+                    attr_dic[name] = idx
+                    idx += 1
+            entry_dic = {}
+            entry_dic["parent_entry"] = onto["parent_entry"]
+            entry_dic["attributes"] = attr_dic
+            self._type_attributes[entry_name] = entry_dic
+
+    def _get_entry_attributes_by_class(
+        self, input_entry_class_name: str
+    ) -> List:
+        """Get type attributes by class name. `input_entry_class_name` should be
+        a fully qualified name of an entry class.
+
+        The `dataclass` module<https://docs.python.org/3/library/dataclasses.html> can add
+        generated special methods to user-defined classes. There is an in-built function
+        called `__dataclass_fields__` that is called on the class object, and it returns
+        all the fields the class contains.
+
+        .. note::
+
+            This function is only applicable to classes decorated as Python
+            `dataclass` since it relies on the `__dataclass_fields__` to find out the attributes.
+
+
+        Args:
+            input_entry_class_name: A fully qualified name of an entry class.
+
+        Returns:
+            A list of attributes corresponding to the input class.
+
+        For example, for Sentence we want to get a list of
```

Comment:
This `Sentence` comes from nowhere, we should explain this is an entry, and use the full qualified name.

Reply:
not fixed. What is a `Sentence`?

Reply:
this is still there.